config/clouddriver.yml

```diff
@@ -101,7 +101,9 @@ dockerRegistry:
     - name: ${providers.dockerRegistry.primaryCredentials.name}
       address: ${providers.dockerRegistry.primaryCredentials.address}
       username: ${providers.dockerRegistry.primaryCredentials.username:}
-      passwordFile: ${providers.dockerRegistry.primaryCredentials.passwordFile}
+      passwordFile: ${providers.dockerRegistry.primaryCredentials.passwordFile:}
+      repositories:
+        - ${providers.dockerRegistry.primaryCredentials.repository}
```

Comment:
Hmm, when I leave this line empty in clouddriver.yml

```yaml
repositories:
  - 
```
to emulate the non-resolved repository name, Spinnaker doesn't query the catalog endpoint. When you ran it, did 

```yaml
repository: ${SPINNAKER_DOCKER_REGISTRY:}
```
cause the catalog endpoint to be queried?